test/server/transformation-controller.spec.js

```diff
@@ -0,0 +1,258 @@
+const expect = require( 'chai' ).expect;
+const request = require( 'supertest' );
+const sinon = require( 'sinon' );
+const communicator = require( '../../app/lib/communicator' );
+const accountModel = require( '../../app/models/account-model' );
+const config = require( '../../app/models/config-model' ).server;
+const surveyModel = require( '../../app/models/survey-model' );
+const userModel = require( '../../app/models/user-model' );
+
+/**
+ * @typedef {import('../../app/models/survey-model').SurveyObject} Survey
+ */
+
+describe( 'Transformation Controller', () => {
+    const basePath = '';
+    const bearer = 'fozzie';
+    const credentials = { bearer };
+    const enketoId = 'surveyZ';
+    const manifestHost = 'http://example.com';
+    const manifestPath = '/manifest';
+    const manifestUrl = `${manifestHost}${manifestPath}`;
+
+    /** @type {import('sinon').SinonSandbox} */
+    let sandbox;
+
+    /** @type {import('express').Application} */
+    let app;
+
+    /** @type {import('http').Server} */
+    let server;
+
+    /** @type {import('../../app/models/account-model').AccountObj */
+    let account;
+
+    /** @type {Survey} */
+    let survey;
+
+    /** @type {Survey} */
+    let surveyWithAccount;
+
+    beforeEach( done => {
+        sandbox = sinon.createSandbox();
+
+        sandbox.stub( config, 'base path' ).get( () => basePath );
+
+        // Stub `_getSurveyParams`
+        survey = {
+            info: {
+                manifestUrl,
+            },
+        };
+
+        sandbox.stub( surveyModel, 'get' ).resolves( survey );
+
+        account = {};
+
+        surveyWithAccount = Object.assign( {}, survey, { account } );
+
+        sandbox.stub( accountModel, 'check' ).resolves( surveyWithAccount );
+
+        // No-op `_checkQuota`
+        sandbox.stub( config, 'account lib' ).get( () => null );
+
+        sandbox.stub( userModel, 'getCredentials' ).resolves( credentials );
+
+        app = require( '../../config/express' );
+
+        server = app.listen( () => done() );
+    } );
+
+    afterEach( done => {
+        sandbox.restore();
+        server.close( done );
+    } );
+
+    const requests = [
+        {
+            description: 'new survey',
+            url: '/transform/xform',
+            body: { xformUrl: 'http://example.com/qwerty' },
+        },
+        {
+            description: 'existing survey',
+            url: `/transform/xform/${enketoId}`,
+        },
+    ];
+
+    requests.forEach( ( { description, url, body } ) => {
```

Comment:
Oh I understand this concern better now that you spelled it out! Yeah, as we've discussed before, I'm similarly cautious about overly "helping" tests. I felt okay with it here because I don't foresee a change which would ever produce conditional behavior to be tested here.